consensus/core/src/core.rs

```diff
@@ -304,6 +307,28 @@ impl Core {
             }
         }
 
+        let leader_authority = &self
+            .context
+            .committee
+            .authority(self.first_round_leader(clock_round))
+            .hostname;
+        self.context
+            .metrics
+            .node_metrics
+            .block_proposal_leader_wait_ms
+            .with_label_values(&[leader_authority])
+            .inc_by(
+                Instant::now()
+                    .saturating_duration_since(self.threshold_clock.get_quorum_ts())
+                    .as_millis() as u64,
+            );
+        self.context
+            .metrics
+            .node_metrics
+            .block_proposal_leader_wait_count
```

Comment:
Just to confirm , this increment will happen on every call, but this might be:
1. either due to block to block process
2. batch block processing

it might still provide signal but it could also be a bit noisy . I guess we can just deploy and see.

Reply:
During catchup I think threshold clock advances very often, also there will be less wait for leader. Let's see if there will be significant noise.